docs/guides/js-waku/use-waku-react.md

```diff
@@ -0,0 +1,356 @@
+---
+title: "Build React DApps Using @waku/react"
+---
+
+The [@waku/react](https://www.npmjs.com/package/@waku/react) package provides components and UI adapters to integrate `js-waku` into React applications effortlessly. This guide provides detailed steps for using `@waku/react` in your project.
+
+## Install the Dependencies
+
+First, set up a project using any [production-grade React framework](https://react.dev/learn/start-a-new-react-project) or use an existing React application. For this guide, we will create a boilerplate using [Create React App (CRA)](https://create-react-app.dev/docs/getting-started):
+
+```mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+```
+
+<Tabs groupId="package-manager">
+<TabItem value="npm" label="npm">
+
+```shell
+npx create-react-app [PROJECT DIRECTORY]
+```
+
+</TabItem>
+<TabItem value="yarn" label="Yarn">
+
+```shell
+yarn create react-app [PROJECT DIRECTORY]
+```
+
+</TabItem>
+</Tabs>
+
+Next, install the `@waku/react` package using your preferred package manager:
+
+<Tabs groupId="package-manager">
+<TabItem value="npm" label="npm">
+
+```shell
+npm install @waku/react @waku/sdk
+```
+
+</TabItem>
+<TabItem value="yarn" label="Yarn">
+
+```shell
+yarn add @waku/react @waku/sdk
+```
+
+</TabItem>
+</Tabs>
+
+## Create a Relay Node
+
+Use the `useCreateRelayNode()` hook to create a Relay Node:
+
+```js title="App.js"
+import { useCreateRelayNode } from "@waku/react";
+
+function App() {
+	// Create and start a Relay Node
+	const { node, error, isLoading } = useCreateRelayNode({
+		options: {
+			defaultBootstrap: true,
+			emitSelf: true,
+		}
+	});
+
+	// "node" is the created Relay Node
+	// "error" captures any error that occurs during node creation
+	// "isLoading" indicates whether the node is still being created
+	// Use these to handle states and render the UI in your application
+}
+```
+
+## Create a Light Node
+
+Use the `useCreateLightNode()` hook to create a Light Node and specify the [protocols](/overview/concepts/protocols) for remote peers:
+
+```js title="App.js"
+import { useCreateLightNode } from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node, error, isLoading } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// "node" is the created Light Node
+	// "error" captures any error that occurs during node creation
+	// "isLoading" indicates whether the node is still being created
+	// Use these to handle states and render the UI in your application
+}
+```
+
+## Create an Encoder and Decoder
+
+Use the `useCreateContentPair()` hook to create a message `encoder` and `decoder` pair:
+
+```js title="App.js"
+import { useCreateContentPair } from "@waku/react";
```

Comment:
I thought we removed this API. I think it may be confusing for the reader that content codec are created together (encoder for write, decoder for read).
This is only useful in a very specific case: a public channel where everyone reads and writes.

I think it limits the creativity of the reader but also confuses them if they intend to do any other sort of communication (broadcast, 1:1, etc).

Instead, I'd recommend one API to create the encoder and one for the decoder. As done in `@waku/core`.

Cc @weboko

Reply:
yes, that's what we agreed on and it is to be addressed in this issue https://github.com/waku-org/js-waku/issues/1060#issuecomment-1447280851 

some of them were closed by the latest release of the `@waku/react` but this one not